src/main/java/duke/logic/Duke.java

```diff
@@ -0,0 +1,47 @@
+package duke.logic;
+
+import duke.DukeException;
+
+/**
+ * Duke class containing all logic components of Duke.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+
+    /**
+     * Constructor for Duke object.
+     *
+     * @param filePath String path of data text file to load data from and save to.
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        tasks = new TaskList();
+
+        try {
+            storage.load(tasks);
+        } catch (DukeException e) {
+            System.out.println("loading error");
+        }
+    }
+
+    public String getResponse(String input) {
+        String[] parsed = Parser.parse(input);
+        return Parser.execute(parsed[0], parsed[1], tasks);
+    }
+
+    /**
+     * Start the Duke task tracker software.
```

Comment:
Not sure if this description of the method is accurate; seems more like saving the list of tasks rather than starting up the software?